src/NumberGuessingGame.jsx

```diff
@@ -11,67 +11,56 @@ function getRandomNumber() {
   return Math.floor(Math.random() * 100) + 1;
 }
 
-const MAX_ATTEMPTS = 5;
+const MAX_ATTEMPTS = 8;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
+function NumberGuessingGame() {
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
+  
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0);
+  const [latestGuess, setLatestGuess] = useState(null);
+  
+  const isCorrectGuess = latestGuess === numberToGuess;
+  const isGameOver =
+    isCorrectGuess || numberOfGuesses === MAX_ATTEMPTS;
+  
+  const handleGuess = (guess) => {
+    setLatestGuess(Number(guess));
+    setNumberToGuess(prev => prev + 1);
+    setNumberOfGuesses(prev=>prev + 1)
   }
+  
+  const handleReset = () => {
+    setLatestGuess(null)
+    setNumberToGuess(getRandomNumber());
+    setNumberOfGuesses(0)
+   }
 
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
-  }
-
-  handleReset() {
-    this.setState({
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    });
-  }
+  return (
+    <div>
+      <h2>I'm thinking of a number from 1 to 100.</h2>
+      <h2>
+        Can you guess the number I am thinking of in {MAX_ATTEMPTS} tries?
+      </h2>
+      <GuessControl onGuess={handleGuess} />
+      {isGameOver && (
+        <GameOver hasWon={isCorrectGuess} onReset={handleReset} />
+      )}
+      {!isGameOver && (
```

Comment:
This should ideally be a ternary operator to make it clear that these two can never both be visible at the same time.